cmd/template.go

```diff
@@ -0,0 +1,56 @@
+/*----------------------------------------------------------------
+ *  Copyright (c) ThoughtWorks, Inc.
+ *  Licensed under the Apache License, Version 2.0
+ *  See LICENSE in the project root for license information.
+ *----------------------------------------------------------------*/
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/getgauge/gauge/logger"
+	"github.com/getgauge/gauge/template"
+	"github.com/spf13/cobra"
+)
+
+var (
+	templateCmd = &cobra.Command{
+		Use:     "template [flags] [args]",
+		Short:   "Change template configurations",
+		Long:    `Change template configurations.`,
+		Example: `  gauge template custom-template ate https://example.org/templates/custom-template.zip`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if templateList || machineReadable {
+				text, err := template.List(machineReadable)
+				if err != nil {
+					logger.Fatalf(true, err.Error())
+				}
+				fmt.Println(text)
+				return
+			}
+			if len(args) == 0 {
+				exit(fmt.Errorf("template command needs argument(s)"), cmd.UsageString())
+			}
+			if len(args) == 1 {
+				text, err := template.Get(args[0])
+				if err != nil {
+					logger.Fatalf(true, err.Error())
+				}
+				logger.Infof(true, text)
+				return
+			}
+			err := template.Update(args[0], args[1])
```

Comment:
This is because we merge, user properties with default properties. So default will always we there. Users can update the defaults but can not remove it. It was by design (at least for `config` properties).
Do we want to change this behavior for templates?

Reply:
Fixed Case 1